ext/geometry/01_geometry.js

```diff
@@ -0,0 +1,2118 @@
+    },
+    {
+      key: "is2D",
+      converter: webidl.converters["boolean"],
+    },
+  ],
+);
+
+const _raw = Symbol("[[raw]]");
+const _writable = Symbol("[[writable]]");
+const _brand = webidl.brand;
+
+class DOMPointReadOnly {
+  [_writable] = false;
+  [_raw];
+
+  constructor(x = 0, y = 0, z = 0, w = 1) {
+    this[_raw] = new Float64Array([
+      webidl.converters["unrestricted double"](x),
+      webidl.converters["unrestricted double"](y),
+      webidl.converters["unrestricted double"](z),
+      webidl.converters["unrestricted double"](w),
+    ]);
+    this[_brand] = _brand;
+  }
+
+  static fromPoint(other = {}) {
+    other = webidl.converters.DOMPointInit(
+      other,
+      "Failed to call 'DOMPointReadOnly.fromPoint'",
+      "Argument 1",
+    );
+    const point = webidl.createBranded(DOMPointReadOnly);
+    point[_writable] = false;
+    point[_raw] = new Float64Array([
+      other.x,
+      other.y,
+      other.z,
+      other.w,
+    ]);
+    return point;
+  }
+
+  get x() {
+    webidl.assertBranded(this, DOMPointReadOnlyPrototype);
+    return this[_raw][0];
+  }
+  get y() {
+    webidl.assertBranded(this, DOMPointReadOnlyPrototype);
+    return this[_raw][1];
+  }
+  get z() {
+    webidl.assertBranded(this, DOMPointReadOnlyPrototype);
+    return this[_raw][2];
+  }
+  get w() {
+    webidl.assertBranded(this, DOMPointReadOnlyPrototype);
+    return this[_raw][3];
+  }
+
+  matrixTransform(matrix = {}) {
+    webidl.assertBranded(this, DOMPointReadOnlyPrototype);
+    const prefix =
+      "Failed to call 'DOMPointReadOnly.prototype.matrixTransform'";
+    if (!ObjectPrototypeIsPrototypeOf(DOMMatrixReadOnlyPrototype, matrix)) {
+      const _matrix = webidl.converters.DOMMatrixInit(
+        matrix,
+        prefix,
+        "Argument 1",
+      );
+      validateAndFixupMatrixDictionary(_matrix, prefix);
+      matrix = {};
+      initMatrixFromDictonary(matrix, _matrix);
+    }
+
+    const point = webidl.createBranded(DOMPoint);
+    point[_writable] = true;
+    point[_raw] = new Float64Array(this[_raw]);
+    op_geometry_premultiply_point_self(matrix[_raw], point[_raw]);
+    return point;
+  }
+
+  toJSON() {
+    webidl.assertBranded(this, DOMPointReadOnlyPrototype);
+    const raw = this[_raw];
+    return {
+      x: raw[0],
+      y: raw[1],
+      z: raw[2],
+      w: raw[3],
+    };
+  }
+
+  [SymbolFor("Deno.privateCustomInspect")](inspect, inspectOptions) {
+    return inspect(
+      createFilteredInspectProxy({
+        object: this,
+        evaluate: ObjectPrototypeIsPrototypeOf(DOMPointReadOnlyPrototype, this),
+        keys: [
+          "x",
+          "y",
+          "z",
+          "w",
+        ],
+      }),
+      inspectOptions,
+    );
+  }
+}
+
+const DOMPointReadOnlyPrototype = DOMPointReadOnly.prototype;
+
+ObjectDefineProperty(DOMPointReadOnlyPrototype, SymbolToStringTag, {
+  value: "DOMPointReadOnly",
+  configurable: true,
+  enumerable: false,
+  writable: false,
+});
```

Comment:
```suggestion
webidl.configureInterface(DOMPointReadOnly);
const DOMPointReadOnlyPrototype = DOMPointReadOnly.prototype;
```

Same for the other 6 interfaces.